SpatialConvolution.lua

```diff
@@ -155,8 +155,8 @@ function SpatialConvolution:accGradParameters(input, gradOutput, scale)
 end
 
 function SpatialConvolution:type(type,tensorCache)
-   self.finput = torch.Tensor()
-   self.fgradInput = torch.Tensor()
+   self.finput = self.finput and self.weight.new()
+   self.fgradInput = self.fgradInput and self.weight.new()
```

Comment:
CPU and GPU version of `SpatialConvolutionMM` uses differently `fgradInput` (GPU assumes that `fgradInput` contains all ones). That's why it was being newly allocating this tensor. I think you should at least resize it to zero. See https://github.com/torch/nn/issues/211